### PR TITLE
Changed wrappedRow to paragraph to fix text wrapping in block explorer (#59)

### PR DESCRIPTION
**Pull request by eulervoid**
## Purpose

This addresses the problems with text wrapping, described in #59. 

(Note: Still needs to be tested on actual transaction data, as i did not have access to the stagenet.)

## Changes

Replaced `wrappedRow` with `paragraph` in the `eventElem` function.

### CODE REVIEW

**Comment by abizjak** [comment]
@eulervoid This is how it looks now, and when copied the line breaks are not copied, which I think is great.

Let me know if this is how you intended it as well.

Screenshot from 2021-09-10 13-46-05

**Comment by eulervoid** [comment]
Looks as intended yes. Thanks for testing!

**Comment by abizjak** [comment]
@eulervoid before we can merge this, can you please bump the version and update the changelog with this bugfix so we can keep track.

**Comment by abizjak** [comment]
We should also check how this affects this issue https://github.com/Concordium/concordium-network-dashboard/issues/40 
Will this break words as well?

**Comment by eulervoid** [comment]
I updated the changelog, the version seemed to be bumped already (1.0.2). 

Regarding #40: on its own it would not break words, but i saw some html attribute for word-break in the code, so in combination with that it might break words as needed.

**Review by limemloh — APPROVED** [review]

File: src/client/elm/Explorer/View.elm [hunk 1]
```diff
@@ -1515,7 +1515,7 @@ wrapAttributes =
 
 eventElem : List (Element msg) -> List (Element msg)
 eventElem es =
-    [ wrappedRow [ width fill ] es ]
+    [ paragraph [ width fill ] es ]
```
Comment (abizjak) [inline src/client/elm/Explorer/View.elm:1518]: Unfortunately this does not always work. It works when there are spaces in the memo, but when there are no spaces we still end up with overflow

Screenshot from 2021-09-13 10-51-23

**Review by abizjak — CHANGES_REQUESTED** [review]

File: src/client/elm/Explorer/View.elm (hunk 1 above)
Comment (abizjak) [reply src/client/elm/Explorer/View.elm:1518]: If it helps this is the block summary response that needs to be displayed correctly
```json
{"finalizationData":{"finalizationIndex":40851,"finalizers":[{"bakerId":0,"signed":true,"weight":3000000000000},{"bakerId":1,"signed":true,"weight":3000000000000},{"bakerId":2,"signed":true,"weight":3000000000000},{"bakerId":3,"signed":true,"weight":3000000000000},{"bakerId":4,"signed":true,"weight":3000000000000},{"bakerId":15,"signed":true,"weight":300292749406},{"bakerId":42,"signed":false,"weight":300143881109}],"finalizationDelay":0,"finalizationBlockPointer":"e784294b2393ceedc1f5c8d9169591bd99dedc0564b8362852a2990f0cebfe00"},"specialEvents":[{"tag":"Mint","foundationAccount":"48RDnoQaCKwy8xteebgLCDdX8qACa3WB5GnfuGMpCDfB2UfdL4","mintPlatformDevelopmentCharge":"138171","mintFinalizationReward":"414508","mintBakingReward":"829017"},{"remainder":"5","tag":"FinalizationRewards","finalizationRewards":[{"amount":"79711","address":"3c1hZst6hfxm5CMPYGmUw7LvMe3Mo1VwgkXrBAb59GSdLzPiSz"},{"amount":"79711","address":"3uyRpq2NPSY4VJn8Qd1uWGcUQTGpCAarHTtqpneiSGqv36Uhna"},{"amount":"79711","address":"4AymJeLgpB7s7bSn6ARm22iAB15m5PpTHaXATBCktnzxvoBowJ"},{"amount":"7974","address":"4C9vCCSLdeTB2DsUZZECReygdLbMR5AACoR1H7EfBfYem7bqLx"},{"amount":"79711","address":"4ESt5DAQY3xi74Pj386otNf2S4fsohDG8Qxyv15hvqvbgFvYvC"},{"amount":"7978","address":"4LRhqFikMSWMzX6ZenkvRVn94H6Zu6HdxmHvap9cuHmMrPraV2"},{"amount":"79711","address":"4M63wWygknpRd1hVuKoirtusZJbWE9wiFe3rXCrZcFgLzXYzh4"}]},{"tag":"BlockReward","bakerReward":"41914","newGASAccount":"42737","baker":"4M63wWygknpRd1hVuKoirtusZJbWE9wiFe3rXCrZcFgLzXYzh4","foundationAccount":"48RDnoQaCKwy8xteebgLCDdX8qACa3WB5GnfuGMpCDfB2UfdL4","oldGASAccount":"1671","foundationCharge":"9220","transactionFees":"92200"}],"updates":{"chainParameters":{"minimumThresholdForBaking":"300000000000","rewardParameters":{"mintDistribution":{"mintPerSlot":7.555665e-10,"bakingReward":0.6,"finalizationReward":0.3},"transactionFeeDistribution":{"gasAccount":0.45,"baker":0.45},"gASRewards":{"chainUpdate":5.0e-3,"accountCreation":2.0e-3,"baker":0.25,"finalizationProof":5.0e-3}},"microGTUPerEuro":{"denominator":1,"numerator":100000000},"foundationAccountIndex":5,"accountCreationLimit":10,"bakerCooldownEpochs":1,"electionDifficulty":2.5e-2,"euroPerEnergy":{"denominator":1000000,"numerator":1}},"keys":{"rootKeys":{"keys":[{"verifyKey":"e0b5fc074eef1178e41f6f2c494bde78b65bd3f21c36db4d41a1e6cb259c3a7d","schemeId":"Ed25519"},{"verifyKey":"1f420ab84863a9ba0472bcb39bfe27a07727f84ace50a6c3f9d6d75e2519febc","schemeId":"Ed25519"},{"verifyKey":"95a5064980b5bc27f3ae96a4831f3c60be82c2c0ee36de3ff028137a99fe823f","schemeId":"Ed25519"},{"verifyKey":"ef3e58b13c43a8439046a774290a8ef0b17d6fe516cea1f48de894a5aec669a4","schemeId":"Ed25519"},{"verifyKey":"830ac28b779ab09d0be9bd15ded7343be1f24df01ccf2e7cffd7d6359500f46c","schemeId":"Ed25519"}],"threshold":3},"level2Keys":{"mintDistribution":{"threshold":3,"authorizedKeys":[17,18,19,20,21]},"addAnonymityRevoker":{"threshold":3,"authorizedKeys":[37,38,39,40,41]},"transactionFeeDistribution":{"threshold":3,"authorizedKeys":[22,23,24,25,26]},"bakerStakeThreshold":{"threshold":3,"authorizedKeys":[32,33,34,35,36]},"microGTUPerEuro":{"threshold":1,"authorizedKeys":[11]},"protocol":{"threshold":2,"authorizedKeys":[5,6,7]},"addIdentityProvider":{"threshold":3,"authorizedKeys":[42,43,44,45,46]},"paramGASRewards":{"threshold":3,"authorizedKeys":[27,28,29,30,31]},"emergency":{"threshold":3,"authorizedKeys":[0,1,2,3,4]},"keys":[{"verifyKey":"97e908ee3a4d8dff49830aacca6bf42997f68afc1851fb5b3d58ae143ba4e147","schemeId":"Ed25519"},{"verifyKey":"f222cc7e2ec7bc1ba7abeca6f503a123abd244292809f04425ad8d72f6ef47ff","schemeId":"Ed25519"},{"verifyKey":"e098e1e55b13988b01e882a7fce5070fd12fa72d409eccebfbad0fb32d6ee057","schemeId":"Ed25519"},{"verifyKey":"58f71ef93d95fd9466aa917d426970c96ff48f9cf1aafb52e4caf427cf6331c8","schemeId":"Ed25519"},{"verifyKey":"9521707ac0ed5b577bc334c25154ab92336a1731bc9bcaefd14cd62bb42dcf96","schemeId":"Ed25519"},{"verifyKey":"38e5321fca15c1f87fe7c1e266f1e85f68a877e3eea11fa45063ad5e4b92e2fb","schemeId":"Ed25519"},{"verifyKey":"0aab1d7d1b81b7526163df81288f645249a0310f8ac84c5e8bc8f7a80be43df1","schemeId":"Ed25519"},{"verifyKey":"74aa4e9dac2bdfea7db5b9fccf240be370746a172a6ae2775b9751d96226dc9d","schemeId":"Ed25519"},{"verifyKey":"4a00ab5a36d4b402ddc4c3160df838f0aba805bb47843e85c8b14971a322f7b0","schemeId":"Ed25519"},{"verifyKey":"a0b17ad7ab31a49168ab4ee479141a2a102d34ee50c1e5abb3b2e78e901760ae","schemeId":"Ed25519"},{"verifyKey":"701376912e8820d404bd93e887d450e115e1dba9c53fdbe4e963ac3be92b508e","schemeId":"Ed25519"},{"verifyKey":"aa5dd0d777e21571ea6bca6fad39cf84e4655643913d83f6ba4da4552f07c3ec","schemeId":"Ed25519"},{"verifyKey":"88766292b604ec67298b817395bb41459575b7eebe22db49a3c87c6e664f67bd","schemeId":"Ed25519"},{"verifyKey":"3f14ba2f24bbb5fb0ec906e7b5080dd104bc7b0ffbf183234db67ab0ccaeec26","schemeId":"Ed25519"},{"verifyKey":"c39edc29fbf11264c60439e202a0a7fa91047fd54808057be50307a12018a38c","schemeId":"Ed25519"},{"verifyKey":"5b85a0fed564bab70b67d201f405fbdf2712498720e2c7de8aa81bce7258319a","schemeId":"Ed25519"},{"verifyKey":"dae562dbf3c082ff4ef58b3308b8858bb2fe105680ad863901776cc5f3131f3d","schemeId":"Ed25519"},{"verifyKey":"5f28823bd52e6bd86d760a8b104bbd4e3b50cf45dfb076f057ab0561756664a7","schemeId":"Ed25519"},{"verifyKey":"d034a8912eeef4c473798e0f790a16612781fb93b286baf20f4553fd07d8e1ca","schemeId":"Ed25519"},{"verifyKey":"e68b20b846c2144d1719f09613ad927055420f7d03282be8c1f0ec32d0a0837e","schemeId":"Ed25519"},{"verifyKey":"bf0c98b711d4bc5f0f2b90d427656234d2d4dc12bca59ae4c931cd9fcd4f66c0","schemeId":"Ed25519"},{"verifyKey":"97e0c06300746dd211f7aa2c29f3628c53036eba3c16a9ad1e116d5b09176ae5","schemeId":"Ed25519"},{"verifyKey":"10de8b005647c3fa2d528c21a1222ff8e2f7f10143e1d1f99f572016c150cdb5","schemeId":"Ed25519"},{"verifyKey":"f8a6f0e1b60caddfdfad28f2ebbe20393a9961ac5a02dd5770dfb4bc2f633d07","schemeId":"Ed25519"},{"verifyKey":"17cb31e51c9197ba1b88231f75ece5b3d33df60898408fac57375971bb68dfad","schemeId":"Ed25519"},{"verifyKey":"661cb015b7ac83ec349f5010d9f73a62f383d9bc14d3256600ba6d26f9042110","schemeId":"Ed25519"},{"verifyKey":"840d332eae5d251094e443a5a0acce58b63c43e72c7ce60add97f211d9b8b198","schemeId":"Ed25519"},{"verifyKey":"55e9a120150626468f029dd343c2b03eb97b8fb6228a5d8772028f8752ad40f2","schemeId":"Ed25519"},{"verifyKey":"4d60696102711d5e536c202a46f149ba8a959a49edf9be8da726a51d2f1185a7","schemeId":"Ed25519"},{"verifyKey":"aa0f8e391f16bd7095b1a3c02ba513a111194d650a7104a627bcbbf6c63d82c1","schemeId":"Ed25519"},{"verifyKey":"8c335e4bc9b8751ac45b15f3b1afea9765051f22393e93f5ff1bfd436d59c4f7","schemeId":"Ed25519"},{"verifyKey":"a5a77daa690591d7e44520303efc843ba359deddcfa89456ab1a7c5a26419696","schemeId":"Ed25519"},{"verifyKey":"e33274cbb31675f479ddff48a163924269cb46b99d814363d66948ccaf3d01f9","schemeId":"Ed25519"},{"verifyKey":"f2eb7342f2dfccbbe15f5f91adda221286b25447a5af8a8db2170d842a5c1350","schemeId":"Ed25519"},{"verifyKey":"cb9b6265f58cafc94bb33210357ee2709f4535b1aab08852d6df7a2421d1cb94","schemeId":"Ed25519"},{"verifyKey":"e26018620b19e90ab8a1a34e6d9d87d851e3f6f9d75f5f5192a30d3aec22f812","schemeId":"Ed25519"},{"verifyKey":"ce2cf1c1082b082abdb4599e15f6303303cc00cb8fe6820c5cda66c443e4144d","schemeId":"Ed25519"},{"verifyKey":"cd735b34b9ebd6d5dcb9cb56040b78e04a5a9449de2473a03887cf6dade2af60","schemeId":"Ed25519"},{"verifyKey":"78f9eac5476dd188177ddf22793c73231ff0abe11614f8373f86f33a2e14fbe6","schemeId":"Ed25519"},{"verifyKey":"802af182e7ba273d1c41441716c21db4773e4635fe1ea27760f33f3ad920eb0c","schemeId":"Ed25519"},{"verifyKey":"a264dd70a3eeac72397cde270806bb36651f25c1971d7a364a87b8d2184ebce0","schemeId":"Ed25519"},{"verifyKey":"951a1a3c5f3907f5706a9a587516cca3c13d8af21abceb4225df66d942d06f02","schemeId":"Ed25519"},{"verifyKey":"285568e67384368b0740309827a295a7c6044db1e7c96f6895c8fc7e2a5b4a38","schemeId":"Ed25519"},{"verifyKey":"bdb46f6cdcf24c5f0f183275c6325ea4034ce3bab1e7a362fce227d37c43e71f","schemeId":"Ed25519"},{"verifyKey":"9b86c0d98a295a23c3c92cb918a48d9b3cf3ab01d5d1d78a036e4a6b7d577c74","schemeId":"Ed25519"},{"verifyKey":"cb7ac7662b4dc4ea902e0d6116dd936ac4e1ace0a86e50059941aba6fd4aefc0","schemeId":"Ed25519"},{"verifyKey":"f69e41a6d9d8417db5497ebe4688b734fb709ee16bed147ded80dfa2930d69db","schemeId":"Ed25519"}],"foundationAccount":{"threshold":3,"authorizedKeys":[12,13,14,15,16]},"electionDifficulty":{"threshold":2,"authorizedKeys":[8,9,10]},"euroPerEnergy":{"threshold":2,"authorizedKeys":[8,9,10]}},"level1Keys":{"keys":[{"verifyKey":"0d57ecfe1f7d941fbc1bc0d001b632052d549cbeab6a966f944e3ad2dac89f6d","schemeId":"Ed25519"},{"verifyKey":"d0d29ce95287f45da364cdd6db512a9eb2bc43704381fc4cf0ffe14540568edc","schemeId":"Ed25519"},{"verifyKey":"94bec66e5b2f69ba94592c98a7facb78033d35c326beecaf263aef536d2a9c2f","schemeId":"Ed25519"},{"verifyKey":"1d13c5e8ca89c4fc0ef7e56e3bcec776490eb364a687c0769540fe2d59fc1b2c","schemeId":"Ed25519"},{"verifyKey":"e9f9715686673769091df75ed0579976b82e83f73cc3559748c2e9627014d6d8","schemeId":"Ed25519"}],"threshold":3}},"updateQueues":{"mintDistribution":{"nextSequenceNumber":1,"queue":[]},"rootKeys":{"nextSequenceNumber":1,"queue":[]},"addAnonymityRevoker":{"nextSequenceNumber":1,"queue":[]},"transactionFeeDistribution":{"nextSequenceNumber":1,"queue":[]},"bakerStakeThreshold":{"nextSequenceNumber":1,"queue":[]},"level2Keys":{"nextSequenceNumber":1,"queue":[]},"microGTUPerEuro":{"nextSequenceNumber":1,"queue":[]},"protocol":{"nextSequenceNumber":2,"queue":[]},"addIdentityProvider":{"nextSequenceNumber":1,"queue":[]},"gasRewards":{"nextSequenceNumber":1,"queue":[]},"foundationAccount":{"nextSequenceNumber":1,"queue":[]},"electionDifficulty":{"nextSequenceNumber":1,"queue":[]},"euroPerEnergy":{"nextSequenceNumber":1,"queue":[]},"level1Keys":{"nextSequenceNumber":1,"queue":[]}}},"transactionSummaries":[{"hash":"1623fb1b65720bbf34b7d9c47818af49e7c4610eaf6f1e241ba1208d1e5d94bb","sender":"4PXJrvKGKb1YZt2Vua3mSDThqUU2EChweuydsFtVFtvDDr585H","cost":"92200","energyCost":922,"result":{"events":[{"amount":"1000000","tag":"Transferred","to":{"address":"48RDnoQaCKwy8xteebgLCDdX8qACa3WB5GnfuGMpCDfB2UfdL4","type":"AddressAccount"},"from":{"address":"4PXJrvKGKb1YZt2Vua3mSDThqUU2EChweuydsFtVFtvDDr585H","type":"AddressAccount"}},{"memo":"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa","tag":"TransferMemo"}],"outcome":"success"},"type":{"contents":"transferWithMemo","type":"accountTransaction"},"index":0}]}
```

**Comment by eulervoid** [comment]
I mocked the http response to test locally and added the necessary html attribute to break long words correctly (see screenshot below). I left some of the mocking code in there as comments, and also couldn't help myself but give the stubs a consistent naming scheme.. i hope thats alright. 

screenshot

File: src/client/elm/Explorer/Stubs.elm [hunk 2]
```diff
@@ -1054,3 +1054,585 @@ getBlockSummaryResponseStub =
   ]
 }
 """
+
+
+getBlockSummaryLongMemoStub =
```
Comment (abizjak) [inline src/client/elm/Explorer/Stubs.elm:1059]: I'm OK with this, but would appreciate it if there was a bit of documentation. The variable name is indicative, but only now that we remember this, so a couple of sentences would be appreciated.

**Review by abizjak — APPROVED** [review]
I believe this will also fix #40 now, but probably not #64, although that can be done in a followup PR with a similar trick (I imagine).

File: src/client/elm/Explorer/Stubs.elm (hunk 2 above)
Comment (eulervoid) [reply src/client/elm/Explorer/Stubs.elm:1062]: I added a comment, but i'm fine with removing the whole thing too if thats preferred.

Comment (abizjak) [reply src/client/elm/Explorer/Stubs.elm:1062]: Looks good.

**Pull request merged**